### ISSUE TITLE
RuleID generic-api-key gets confused by key_algorithm

### ISSUE THREAD

**Issue by dhoppe** [zricethezav/gitleaks#1053]
**Describe the bug**
I think the following regular expression is too sensitive and should only act on ```key```, not the ```key_algorithm```.
https://github.com/zricethezav/gitleaks/blob/master/config/gitleaks.toml#L482-L490

**To Reproduce**
Create a file that contains the following argument:
```
key_algorithm = "EC_secp384r1"
```

**Expected behavior**
The argument ```key_algorithm``` should be ignored since it does not contain any sensitive information. 

**Screenshots**
```
Finding:     key_algorithm = "REDACTED"
Secret:      REDACTED
RuleID:      generic-api-key
Entropy:     3.584963
File:        _common/confluence/acm.hcl
Line:        18
Fingerprint: _common/confluence/acm.hcl:generic-api-key:18
```

**Basic Info (please complete the following information):**
 - OS: macOS Ventura 13.0.1
 - Gitleaks Version: v8.15.2

**Additional context**
```
$ grep key_algorithm _common/confluence/acm.hcl
18:  key_algorithm = "EC_secp384r1"
```

cc @zricethezav

**Comment by dhoppe** [comment]
I fixed this by adding ```# gitleaks:allow``` but I think the default config should be updated since ```key_algorithm``` does not contain a secret.

**Comment by weineran** [comment]
@dhoppe You are right that it's not sensitive info, but given the regex-based approach that gitleaks takes to secret detection, I'm not sure how to change the rule to fix false positives like this one.

Anytime the identifier contains the string "key", this rule is likely to fire. Example:
```
hankey_pankey = "nothing sensitive here!"
```

One idea is that we could add all the key algorithms listed [here](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-acm/enums/keyalgorithm.html) to the `stopwords` for the `generic-api-key` rule [here](https://github.com/zricethezav/gitleaks/blob/master/config/gitleaks.toml#L495). This is only a partial solution to the broader problem, but it would address your specific false positive...

@zricethezav Thoughts?
